### ISSUE TITLE
正常 vs 异常音频特征向量余弦相似度偏高

### ISSUE THREAD

**Issue by gouxingtianxia** [jianganbai/FISHER#6]
您好，感谢开源 FISHER 项目！我在使用示例 Inference 代码测试时，遇到正常音频和异常音频特征向量余弦相似度偏高的问题：
- 测试过程：如下两个测试文件，用 FISHER 提取特征向量，并计算余弦相似度。
- 测试结果：相似度 ≈ 0.7155，与两个正常声音之间的相似度相当。

- 请教以下问题

1. 测试文件中的两个波形理论上是否适合使用FISHER进行异常检测？
2. 如果希望让测试文件中两个波形特性向量相似度显著降低（例如 < 0.5），应该如何做，是否需要使用异常音频数据进行训练？
3. FISHER 提取的特征向量是否具有可描述性，如果可描述的话，主要包含哪些信息？

- 测试文件：

    **normal-0-channel_0.wav（正常声音）**

    [normal-0-channel_0.wav](https://github.com/user-attachments/files/23746727/normal-0-channel_0.wav)

    <img width="650" height="129" alt="Image" src="https://github.com/user-attachments/assets/f260ecf2-5ecc-463c-bae0-94682107e11e" />

   **abnormal-5-channel_0.wav（异常声音）**

    [abnormal-5-channel_0.wav](https://github.com/user-attachments/files/23746737/abnormal-5-channel_0.wav)

   <img width="649" height="127" alt="Image" src="https://github.com/user-attachments/assets/c8c52963-1af8-40a3-91ac-6821ee54a00d" />

- 测试代码
`
def process_audio(wav_path, model):
    wav, sr = torchaudio.load(wav_path)
    wav = wav - wav.mean()
    
    STFT = torchaudio.transforms.Spectrogram(
        n_fft=25 * sr // 1000,
        win_length=None,
        hop_length=10 * sr // 1000,
        power=1,
        center=False
    )
    spec = torch.log(torch.abs(STFT(wav)) + 1e-10)
    spec = spec.transpose(-2, -1)  # [1, time, freq]
    spec = (spec + 3.017344307886898) / (2.1531635155379805 * 2)
    
    if spec.shape[-2] > 1024:
        spec = spec[:, :1024]
    
    if spec.shape[-1] < model.cfg.band_width:
        spec = F.pad(spec, (0, model.cfg.band_width - spec.shape[-1]))
    
    spec = spec.unsqueeze(1).cuda()
    
    with torch.no_grad():
        with torch.autocast('cuda'):
            repre = model.extract_features(spec)
    
    return repre

model_path = '/home/nvidia/llm/FISHER/FISHER-small.pt'
model = FISHER.from_pretrained(model_path)
model = model.cuda()
model.eval()

wav_path_1 = '/home/nvidia/llm/FISHER/wav/normal-0-channel_0.wav'
repre1 = process_audio(wav_path_1, model)
print("shape:", repre1.shape)

wav_path_2 = '/home/nvidia/llm/FISHER/wav/abnormal-5-channel_0.wav'
repre2 = process_audio(wav_path_2, model)
print("shape2:", repre2.shape)

cos_sim = F.cosine_similarity(repre1, repre2, dim=1)
print("余弦相似度:", cos_sim)

`
